Android/Tapp/app/src/main/java/app/anish/com/tapp/linkedin_auth/LinkedInWebLoginFlow.java

```diff
@@ -0,0 +1,92 @@
+package app.anish.com.tapp.linkedin_auth;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.DisplayMetrics;
+import android.widget.Toast;
+
+import com.linkedin.platform.AccessToken;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import app.anish.com.tapp.dialogs.LinkedInWebViewLoginDialog;
+import app.anish.com.tapp.utils.HttpUtils;
+
+/**
+ * Login flow through through Web Dialog
+ */
+
+public final class LinkedInWebLoginFlow extends LinkedInLoginFlow {
+
+    private Context context;
+    private FlowCompletionListener listener;
+
+    public LinkedInWebLoginFlow(Context context, FlowCompletionListener listener) {
+        this.context = context;
+        this.listener = listener;
+    }
+
+
+    @Override
+    public void startFlow() {
+        Dialog dialog = new LinkedInWebViewLoginDialog(context, new LinkedInWebViewLoginDialog.AuthListener() {
+            @Override
+            public void onAuthSuccess(AccessToken accessToken) {
+                getUserInfo(accessToken);
+            }
+
+            @Override
+            public void onAuthError(String error) {
+                listener.onFailure(new Exception(error));
+            }
+        });
+
+        DisplayMetrics metrics = context.getResources().getDisplayMetrics();
```

Comment:
This two line section may end up looking horrible on different devices/orientations. Most of the time, you can solve these sort of display issues in a way that is more compatible with more devices/orientations.